cluster-autoscaler/main.go

```diff
@@ -252,6 +253,12 @@ func createAutoscalingOptions() config.AutoscalingOptions {
 	if *maxDrainParallelismFlag > 1 && !*parallelDrain {
 		klog.Fatalf("Invalid configuration, could not use --max-drain-parallelism > 1 if --parallel-drain is false")
 	}
+
+	// In order to avoid inconsistent deletion thresholds for the legacy planner and the new actuator,
+	// we set the deletion threshold to the maximum of the two flags that are used to filter down deletion candidates.
+	maxDeletionSize := int(math.Max(float64(*maxEmptyBulkDeleteFlag), float64(*maxScaleDownParallelismFlag)))
+	*maxEmptyBulkDeleteFlag = maxDeletionSize
+	*maxScaleDownParallelismFlag = maxDeletionSize
```

Comment:
This whole piece of code should be removed when we properly deprecate maxEmptyBulkDelete